waspc/src/Wasp/Util/Json.hs

```diff
@@ -16,3 +25,12 @@ parseJsonWithComments jsonStr = do
   case exitCode of
     ExitSuccess -> return $ decodeFromString response
     _exitFailure -> return $ Left stderr
+
+updateJsonFile :: (Value -> Value) -> Path' Abs (File a) -> IO (Either String ())
+updateJsonFile updateFn jsonFilePath = runExceptT $ do
+  jsonContent <- ExceptT $ eitherDecode <$> IOUtil.readFileBytes jsonFilePath
+  let updatedContent = updateFn jsonContent
+  liftIO $ writeJsonValue jsonFilePath updatedContent
+
+writeJsonValue :: Path' Abs (File f) -> Value -> IO ()
+writeJsonValue file = IOUtil.writeFileBytes file . encode
```

Comment:
I thought about whether I should put this here or in IOUtil, but I decided this makes more sense in the end.